src/app/components/Cart.tsx

```diff
@@ -0,0 +1,57 @@
+'use client'
+import { useCart } from "../contexts/CartContext"
+import Image from "next/image";
+
+const Cart = () => {
+    const { cartItems,removeFromCart } = useCart();
+  
+    return (
+        <div className="flex flex-col w-2/3 bg-gray-100 text-gray-800">
+            <header className="bg-white py-4 shadow">
+                <h1 className="text-2xl font-bold text-center">Shopping Cart</h1>
+            </header>
+            
+            <main className="flex-1 px-4 py-8 md:px-8">
+                <section className="max-w-3xl mx-auto">
+                    <ul className="space-y-4">
+                        {cartItems.reduce((aggregatedItems, item) => {
+                            const existingItem = aggregatedItems.find(uniqueItem => uniqueItem.id === item.id);
+
+                            if (existingItem) {
+                            existingItem.quantity += item.quantity;
+                            } else {
+                                aggregatedItems.push({ ...item });
+                        
+                            }
+
+                            return aggregatedItems;
```

Comment:
This logic is to find duplicated values, right? I think you should handle this when adding product to cart, not when consuming items.